librato/__init__.py

```diff
@@ -421,10 +421,56 @@ def list_alerts(self, active_only=True, **query_props):
         else:
             return alerts
 
+    def list_all_alerts(self, active_only=True, **query_props):
+        """List all avaliable alerts.  In addition to the normal query_props,
+        active_only=<bool> can be passed in to filter out inactive alerts
+
+        :type active_only: bool
+        :param active_only: Whether to only return active alerts, or all alerts
+
+        """
+        if 'length' not in query_props:
+            query_props['length'] = 100
+        if 'offset' not in query_props:
+            query_props['offset'] = 0
+        page_size = query_props['length']
+        while True:
+            # Get all alerts from list_alerts, and filter them below so that
+            # we can determine when to terminate in the presence of pagination.
+            metric_list = self.list_alerts(**query_props)
+            for m in metric_list:
+                if active_only:
+                    if not m.active:
+                        continue
+                    yield m
+                else:
+                    yield m
+            query_props['offset'] += page_size
+            if len(metric_list) < page_size:
+                break
+
+
     def list_services(self, **query_props):
         resp = self._mexe("services", query_props=query_props)
         return self._parse(resp, "services", Service)
 
+    def list_all_services(self, **query_props):
+        if 'length' not in query_props:
+            query_props['length'] = 100
+        if 'offset' not in query_props:
+            query_props['offset'] = 0
+        page_size = query_props['length']
+        while True:
+            # Get all alerts from list_alerts, and filter them below so that
```

Comment:
Why not just fold this into `list_services` and `yield` each result?  You're encoding a parameter (i.e. "limit") into the function.

    `list_alerts(limit=None)` (same as "list_all_alerts", should be the default)
    `list_alerts(limit=100)`

You can even provide a `page_size` parameter if you want "flexibility" over actual request paging.